plugins/hdf/io/HdfReader.cpp

```diff
@@ -124,17 +128,36 @@ void HdfReader::addArgs(ProgramArgs& args)
 
 void HdfReader::initialize()
 {
-    log()->get(LogLevel::Info) << "***JSON TESTING***" << std::endl;
-    log()->get(LogLevel::Info) << m_pathDimMap << std::endl;
-    log()->get(LogLevel::Info) << "----------------" << std::endl;
     for(const auto& [key, value] : m_pathDimMap.items()) {
         log()->get(LogLevel::Info) << "Key: " << key << ", Val: " << value <<std::endl;
     }
+    validateMap();
 
     // ICESat-2 datasets are EPSG:7912
     setSpatialReference(SpatialReference("EPSG:7912"));
 }
 
+void HdfReader::validateMap()
+{
+    log()->get(LogLevel::Info) << "**JSON map**" << std::endl;
+    log()->get(LogLevel::Info) << m_pathDimMap << std::endl;
+
+    if(!m_pathDimMap.is_object()) {
+        throw pdal_error("Option 'map' must be a JSON object, not a " +
+            std::string(m_pathDimMap.type_name()));
+    }
+    for(auto& [dimName, datasetName] : m_pathDimMap.items()) {
```

Comment:
This syntax isn't valid before C++17.  You'll have to get back the actual object and split it apart later.  If it builds on your system, you're getting lucky.